### PR TITLE
Upgrading bicep templates #886

### LINKED ISSUE #886: Upgrading bicep templates
During the deployment, there are several BICEP warnings that alert to changes on the bicep specifications and that there are newer versions.

**DoD**
- Review all bicep modules from the sample and upgrade to the newest version
- Test e2e sample to verify that there is no breaking change
- Create a table in the ReadMe specifying what is the current version per each bicep module and in which date was upgraded in the sample

### PR DESCRIPTION
**Pull request by raafatzarka**
# Type of PR
- Code changes

## Purpose
Review all bicep modules from the sample and upgrade to the newest version

## Author pre-publish checklist

- [ ] Added test to prove my fix is effective or new feature works
- [ ] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps

- tested the deployment using Dev Container
- All resources have been deployed successfully

## Issues Closed or Referenced

- Closes #886 

- References #995

### CODE REVIEW

**Review by thesqlpro — APPROVED** [review]
looks good

**Comment by ydaponte** [comment]
@raafatzarka - it seems you still have conflicts with the base branch.

File: modern-data-warehouse-dataops
Comment (ydaponte) [inline modern-data-warehouse-dataops:1]: What is this file?

**Review by ydaponte — CHANGES_REQUESTED** [review]
Minor question on an added file and conflicts must be resolved.

**Review by ydaponte — APPROVED** [review]
Approving! Thanks!

**Pull request merged**